src/ducks/stripeConnectAccount.duck.js

```diff
@@ -0,0 +1,265 @@
+          ? e.apiErrors[0].meta.stripeMessage
+          : null;
+      log.error(err, 'create-stripe-account-failed', { stripeMessage });
+      throw e;
+    });
+};
+
+// This function is used for updating the bank account token
+// but could be expanded to other information as well.
+//
+// If the Stripe account has been created with account token,
+// you need to use account token also to update the account.
+// By default the account token will not be used.
+// See API reference for more information:
+// https://www.sharetribe.com/api-reference/?javascript#update-stripe-account
+export const updateStripeAccount = params => (dispatch, getState, sdk) => {
+  const bankAccountToken = params.bankAccountToken;
+
+  dispatch(stripeAccountUpdateRequest());
+  return sdk.stripeAccount
+    .update(
+      { bankAccountToken, requestedCapabilities: ['card_payments', 'transfers'] },
+      { expand: true }
+    )
+    .then(response => {
+      const stripeAccount = response.data.data;
+      dispatch(stripeAccountUpdateSuccess(stripeAccount));
+      return stripeAccount;
+    })
+    .catch(err => {
+      const e = storableError(err);
+      dispatch(stripeAccountUpdateError(e));
+      const stripeMessage =
+        e.apiErrors && e.apiErrors.length > 0 && e.apiErrors[0].meta
+          ? e.apiErrors[0].meta.stripeMessage
+          : null;
+      log.error(err, 'update-stripe-account-failed', { stripeMessage });
+      throw e;
+    });
+};
+
+export const fetchStripeAccount = params => (dispatch, getState, sdk) => {
+  dispatch(stripeAccountFetchRequest());
+
+  return sdk.stripeAccount
+    .fetch()
+    .then(response => {
+      const stripeAccount = response.data.data;
+      dispatch(stripeAccountFetchSuccess(stripeAccount));
+      return stripeAccount;
+    })
+    .catch(err => {
+      const e = storableError(err);
+      dispatch(stripeAccountFetchError(e));
+      const stripeMessage =
+        e.apiErrors && e.apiErrors.length > 0 && e.apiErrors[0].meta
+          ? e.apiErrors[0].meta.stripeMessage
+          : null;
+      log.error(err, 'fetch-stripe-account-failed', { stripeMessage });
+      throw e;
+    });
+};
+
+export const getStripeConnectAccountLink = params => (dispatch, getState, sdk) => {
+  const { failureURL, successURL, type } = params;
+  dispatch(getAccountLinkRequest());
+
+  return sdk.stripeAccountLinks
+    .create({
+      failureURL,
+      successURL,
+      type,
+      collect: 'currently_due',
+    })
+    .then(response => {
+      // Return the account link
+      return response.data.data.attributes.url;
+    })
+    .catch(err => {
+      const e = storableError(err);
+      dispatch(getAccountLinkError(e));
+      const stripeMessage =
+        e.apiErrors && e.apiErrors.length > 0 && e.apiErrors[0].meta
+          ? e.apiErrors[0].meta.stripeMessage
+          : null;
+      log.error(err, 'get-stripe-account-link-failed', { stripeMessage });
+      throw e;
+    });
```

Comment:
This `.catch` is repeating a few times here. Would it make sense to extract (some of) it for reuse?

Reply:
I decided not to do this right now but I think this is a good suggestion. I added a card to Clubhouse. It might be useful to check also stripe.duck.js in case there is also something similar so this could be extracted to a helper function.